SDWebImage/SDWebImageDefine.h

```diff
@@ -12,6 +12,113 @@ typedef void(^SDWebImageNoParamsBlock)(void);
 typedef NSString * SDWebImageContextOption NS_STRING_ENUM;
 typedef NSDictionary<SDWebImageContextOption, id> SDWebImageContext;
 
+typedef NS_OPTIONS(NSUInteger, SDWebImageOptions) {
```

Comment:
This is just code moving :)

Because that `SDWebImageLoader` protocol should not import `SDWebImageDownloader`, or it will cause a cycle import issue.

I can seperate this into another PR after some 5.x feature merged to avoid future merge conflict. Not a big change.

Reply:
This is already been done in current 5.x branch.